tag/context.go

```diff
@@ -22,7 +22,7 @@ func FromContext(ctx context.Context) *Map {
 	// The returned tag map shouldn't be mutated.
 	ts := ctx.Value(mapCtxKey)
 	if ts == nil {
-		return newMap(0)
```

Comment:
I thought that the purpose of FromContext is to create an empty map if none exists, and for convenience that way we don't have to always check `if ts != nil`?

Reply:
FromContext should return nil. This gives the user the signal, for example, whether they should Encode and propagate the tags.

OTOH, I made the map methods defensive and they handle the nil value.